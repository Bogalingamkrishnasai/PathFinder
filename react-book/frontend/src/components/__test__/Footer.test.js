import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Footer from '../Footer';
import '@testing-library/jest-dom';

describe('Footer Component', () => {
  test('renders footer with sections and links', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    );
    
    // Check for section headings
    expect(screen.getByText('Pathfinder')).toBeInTheDocument();
    expect(screen.getByText('Career Resources')).toBeInTheDocument();
    expect(screen.getByText('Connect')).toBeInTheDocument();
    
    // Check for navigation links
    expect(screen.getByText('About Us')).toBeInTheDocument();
    expect(screen.getByText('Career Blog')).toBeInTheDocument();

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders social media links', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    );
    
    const socialLinks = screen.getAllByRole('link').filter(link => 
      link.href.includes('twitter.com') || 
      link.href.includes('linkedin.com') || 
      link.href.includes('facebook.com')
    );
    expect(socialLinks).toHaveLength(3);

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders copyright notice', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    );
    
    expect(screen.getByText(/Â© 2024 Pathfinder/i)).toBeInTheDocument();

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });

  // Dedicated snapshot test for full footer rendering
  test('matches footer snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});