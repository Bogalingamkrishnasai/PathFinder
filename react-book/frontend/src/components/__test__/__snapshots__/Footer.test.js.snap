// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches footer snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="footer-content"
    >
      <div
        class="footer-section"
      >
        <h4>
          Pathfinder
        </h4>
        <ul
          class="footer-links"
        >
          <li>
            <a
              href="/about"
            >
              About Us
            </a>
          </li>
          <li>
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              href="/privacy"
            >
              Privacy Policy
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-section"
      >
        <h4>
          Career Resources
        </h4>
        <ul
          class="footer-links"
        >
          <li>
            <a
              href="/blog"
            >
              Career Blog
            </a>
          </li>
          <li>
            <a
              href="/guides"
            >
              Guides
            </a>
          </li>
          <li>
            <a
              href="/tools"
            >
              Career Tools
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-section"
      >
        <h4>
          Connect
        </h4>
        <div
          class="social-links"
        >
          <a
            href="https://twitter.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-twitter"
            />
          </a>
          <a
            href="https://linkedin.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-linkedin"
            />
          </a>
          <a
            href="https://facebook.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-facebook"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="footer-bottom"
    >
      © 2024 Pathfinder. All rights reserved.
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Footer Component renders copyright notice 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="footer-content"
    >
      <div
        class="footer-section"
      >
        <h4>
          Pathfinder
        </h4>
        <ul
          class="footer-links"
        >
          <li>
            <a
              href="/about"
            >
              About Us
            </a>
          </li>
          <li>
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              href="/privacy"
            >
              Privacy Policy
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-section"
      >
        <h4>
          Career Resources
        </h4>
        <ul
          class="footer-links"
        >
          <li>
            <a
              href="/blog"
            >
              Career Blog
            </a>
          </li>
          <li>
            <a
              href="/guides"
            >
              Guides
            </a>
          </li>
          <li>
            <a
              href="/tools"
            >
              Career Tools
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-section"
      >
        <h4>
          Connect
        </h4>
        <div
          class="social-links"
        >
          <a
            href="https://twitter.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-twitter"
            />
          </a>
          <a
            href="https://linkedin.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-linkedin"
            />
          </a>
          <a
            href="https://facebook.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-facebook"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="footer-bottom"
    >
      © 2024 Pathfinder. All rights reserved.
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Footer Component renders footer with sections and links 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="footer-content"
    >
      <div
        class="footer-section"
      >
        <h4>
          Pathfinder
        </h4>
        <ul
          class="footer-links"
        >
          <li>
            <a
              href="/about"
            >
              About Us
            </a>
          </li>
          <li>
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              href="/privacy"
            >
              Privacy Policy
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-section"
      >
        <h4>
          Career Resources
        </h4>
        <ul
          class="footer-links"
        >
          <li>
            <a
              href="/blog"
            >
              Career Blog
            </a>
          </li>
          <li>
            <a
              href="/guides"
            >
              Guides
            </a>
          </li>
          <li>
            <a
              href="/tools"
            >
              Career Tools
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-section"
      >
        <h4>
          Connect
        </h4>
        <div
          class="social-links"
        >
          <a
            href="https://twitter.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-twitter"
            />
          </a>
          <a
            href="https://linkedin.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-linkedin"
            />
          </a>
          <a
            href="https://facebook.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-facebook"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="footer-bottom"
    >
      © 2024 Pathfinder. All rights reserved.
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Footer Component renders social media links 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="footer-content"
    >
      <div
        class="footer-section"
      >
        <h4>
          Pathfinder
        </h4>
        <ul
          class="footer-links"
        >
          <li>
            <a
              href="/about"
            >
              About Us
            </a>
          </li>
          <li>
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              href="/privacy"
            >
              Privacy Policy
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-section"
      >
        <h4>
          Career Resources
        </h4>
        <ul
          class="footer-links"
        >
          <li>
            <a
              href="/blog"
            >
              Career Blog
            </a>
          </li>
          <li>
            <a
              href="/guides"
            >
              Guides
            </a>
          </li>
          <li>
            <a
              href="/tools"
            >
              Career Tools
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-section"
      >
        <h4>
          Connect
        </h4>
        <div
          class="social-links"
        >
          <a
            href="https://twitter.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-twitter"
            />
          </a>
          <a
            href="https://linkedin.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-linkedin"
            />
          </a>
          <a
            href="https://facebook.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-facebook"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="footer-bottom"
    >
      © 2024 Pathfinder. All rights reserved.
    </div>
  </footer>
</DocumentFragment>
`;
